morph/react/block-list.js

```diff
@@ -16,10 +18,50 @@ export function enter(node, parent, state) {
   state.render.push(
     `{Array.isArray(${value}) && ${value}.map((item, index, list) => `
   )
+
+  if (parent.viewPath) {
+    let itemDataContextName =
+      getProp(node, 'itemDataContextName') ||
+      defaultItemDataContextName(state, node, from.value)
+    state.render.push(
+      `<fromData.DataProvider
```

Comment:
forgot to say that we'll need a key prop here to id the different elements, otherwise react will give out, we're doing it on the list item now so it'll need to be moved here - see `morph/react/block-in-list.js` and `morph/react-native/block-wrap.js`

Reply:
makes complete sense, sorry I missed that my bad